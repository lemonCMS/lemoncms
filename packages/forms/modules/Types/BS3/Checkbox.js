import React from "react";
import PropTypes from "prop-types";
import { FormControl } from "react-bootstrap-v3";
import context from "../../decorators/context";
import fieldGroup from "./fieldGroup";

const Checkbox = props => {
  const { input } = props;

  return <FormControl {...input} type={"password"} />;
};

Checkbox.propTypes = {
  input: PropTypes.oneOfType([PropTypes.object]),
  placeholder: PropTypes.string,
  label: PropTypes.string,
  help: PropTypes.string,
  computedInvalid: PropTypes.bool.isRequired
};

Checkbox.defaultProps = {
  input: {},
  label: null,
  help: null,
  placeholder: null
};

export default context(fieldGroup(Checkbox));
