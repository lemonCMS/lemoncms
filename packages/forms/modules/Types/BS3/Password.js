import React from "react";
import PropTypes from "prop-types";
import {
  FormGroup,
  ControlLabel,
  FormControl,
  HelpBlock
} from "react-bootstrap-v3";
import context from "../../decorators/context";
import field from "../../decorators/field";

const Password = props => {
  const { input, placeholder, computedInvalid, label, help } = props;

  return (
    <FormGroup validationState={computedInvalid ? "error" : null}>
      {label && <ControlLabel>{label}</ControlLabel>}
      <FormControl type="password" placeholder={placeholder} {...input} />
      <FormControl.Feedback />
      {help && <HelpBlock>{help}</HelpBlock>}
    </FormGroup>
  );
};

Password.propTypes = {
  input: PropTypes.oneOfType([PropTypes.object]),
  placeholder: PropTypes.string,
  label: PropTypes.string,
  help: PropTypes.string,
  computedInvalid: PropTypes.bool.isRequired
};

Password.defaultProps = {
  input: {},
  label: null,
  help: null,
  placeholder: null
};

export default context(field(Password));
