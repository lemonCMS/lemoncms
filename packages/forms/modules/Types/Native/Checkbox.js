import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import context from '../../decorators/context';
import field from '../../decorators/field';

const Checkbox = props => {
  const {
    input, isDisabled, multiple, children, formControl, computedInvalid
  } = props;

  const radioList = Array.isArray(children) ? children : [children];

  const onChange = (event, key, option) => {
    if (multiple) {
      const newValue = (input.value instanceof Array ? input.value : [input.value]).filter(Boolean);
      if (event.target.checked) {
        newValue.push(option.props.value);
      } else {
        newValue.splice(newValue.indexOf(option.props.value), 1);
      }
      return input.onChange(newValue);
    }
    return input.onChange(event.target.checked ? option.props.value : false);
  };

  const list = radioList.map((option, key) => {
    const name = multiple ? `${input.name}[${key}]` : input.name;
    const value = (Array.isArray(input.value) ? input.value : [input.value]).filter(val => String(val));
    const checked = value.indexOf(option.props.value) !== -1;

    return (
      <div className="form-check" key={option.props.value}>
        <input
          type="checkbox"
          name={name}
          disabled={isDisabled}
          id={`${input.name}_${key}`}
          value={option.props.value}
          className={classNames('form-check-label')}
          checked={checked}
          onChange={event => onChange(event, key, option)}
        />{' '}
        <label htmlFor={`${input.name}_${key}`} className={classNames('form-check-label')}>
          {option.props.children}
        </label>
      </div>
    );
  });

  return <div className={classNames('ffc-input', { 'is-invalid': computedInvalid }, formControl)}>{list}</div>;
};

Checkbox.propTypes = {
  input: PropTypes.oneOfType([PropTypes.object]),
  children: PropTypes.oneOfType([PropTypes.object, PropTypes.array]).isRequired,
  formControl: PropTypes.string,
  multiple: PropTypes.bool,
  isDisabled: PropTypes.bool,
  computedInvalid: PropTypes.bool.isRequired
};

Checkbox.defaultProps = {
  input: {},
  formControl: null,
  multiple: false,
  isDisabled: false
};

export default context(field(Checkbox));
