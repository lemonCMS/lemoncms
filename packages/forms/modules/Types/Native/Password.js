import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import context from '../../decorators/context';
import field from '../../decorators/field';

const Password = props => {
  const {
    input, placeholder, formControl, computedInvalid
  } = props;
  return (
    <input
      type={'password'}
      placeholder={placeholder}
      className={classNames('ffc-input', { 'is-invalid': computedInvalid }, formControl)}
      {...input}
    />
  );
};

Password.propTypes = {
  input: PropTypes.oneOfType([PropTypes.object]),
  placeholder: PropTypes.string,
  formControl: PropTypes.string,
  computedInvalid: PropTypes.bool.isRequired
};

Password.defaultProps = {
  input: {},
  placeholder: null,
  formControl: null
};

export default context(field(Password));
