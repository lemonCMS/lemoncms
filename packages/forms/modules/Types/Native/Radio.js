import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import context from '../../decorators/context';
import field from '../../decorators/field';

const Radio = props => {
  const {
    input, children, formControl, computedInvalid
  } = props;

  const radioList = Array.isArray(children) ? children : [children];

  const onChange = (event, key, option) => {
    input.onChange(option.props.value);
  };

  const list = radioList.map((option, key) => (
    <div className="form-check" key={option.props.value}>
      <input
        type="radio"
        name={`${input.name}`}
        id={`${input.name}_${key}`}
        value={option.props.value}
        className={classNames('form-check-label')}
        onChange={event => onChange(event, key, option)}
        checked={String(input.value) === String(option.props.value)}
      />{' '}
      <label htmlFor={`${input.name}_${key}`} className={classNames('form-check-label')}>
        {option.props.children}
      </label>
    </div>
  ));

  return <div className={classNames('ffc-input', { 'is-invalid': computedInvalid }, formControl)}>{list}</div>;
};

Radio.propTypes = {
  input: PropTypes.oneOfType([PropTypes.object]),
  children: PropTypes.oneOfType([PropTypes.object, PropTypes.array]).isRequired,
  formControl: PropTypes.string,
  computedInvalid: PropTypes.bool.isRequired
};

Radio.defaultProps = {
  input: {},
  formControl: null
};

export default context(field(Radio));
