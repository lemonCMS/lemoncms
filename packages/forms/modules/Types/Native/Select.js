import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import context from '../../decorators/context';
import field from '../../decorators/field';

const Select = props => {
  const {
    input, children, placeholder, formControl, computedInvalid
  } = props;

  const {value, ...rest} = input;


  const options = Array.isArray(children) ? children : [children];
  const list = options.map((option, key) => (
    <option
      key={key}
      value={option.props.value}
      selected={String(input.value) === String(option.props.value)}
    >
      {option.props.children}
    </option>
  ));

  return (
    <select
      placeholder={placeholder}
      className={classNames('ffc-input', { 'is-invalid': computedInvalid }, formControl)}
      {...rest}
    >
      {list}
    </select>
  );
};

Select.propTypes = {
  input: PropTypes.oneOfType([PropTypes.object]),
  children: PropTypes.oneOfType([PropTypes.object, PropTypes.array]).isRequired,
  placeholder: PropTypes.string,
  formControl: PropTypes.string,
  computedInvalid: PropTypes.bool.isRequired
};

Select.defaultProps = {
  input: {},
  placeholder: null,
  formControl: null
};

export default context(field(Select));
