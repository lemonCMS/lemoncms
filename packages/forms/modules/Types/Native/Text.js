import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import context from '../../decorators/context';
import field from '../../decorators/field';

const Text = props => {
  const {
    input, placeholder, formControl, computedInvalid
  } = props;
  return (
    <input
      type="text"
      placeholder={placeholder}
      className={classNames('ffc-input', { 'is-invalid': computedInvalid }, formControl)}
      {...input}
    />
  );
};

Text.propTypes = {
  input: PropTypes.oneOfType([PropTypes.object]),
  placeholder: PropTypes.string,
  formControl: PropTypes.string,
  computedInvalid: PropTypes.bool.isRequired
};

Text.defaultProps = {
  input: {},
  placeholder: null,
  formControl: null
};

export default context(field(Text));
