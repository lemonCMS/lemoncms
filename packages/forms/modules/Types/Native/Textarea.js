import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import context from '../../decorators/context';
import field from '../../decorators/field';

const Textarea = props => {
  const {
    input, placeholder, formControl, computedInvalid, rows
  } = props;
  const {value, ...rest} = input;
  return (
    <textarea
      placeholder={placeholder}
      className={classNames('ffc-input', { 'is-invalid': computedInvalid }, formControl)}
      rows={rows}
      {...rest}
    />
  );
};

Textarea.propTypes = {
  input: PropTypes.oneOfType([PropTypes.object]),
  placeholder: PropTypes.string,
  formControl: PropTypes.string,
  computedInvalid: PropTypes.bool.isRequired,
  rows: PropTypes.number
};

Textarea.defaultProps = {
  input: {},
  placeholder: null,
  formControl: null,
  rows: 10
};

export default context(field(Textarea));
