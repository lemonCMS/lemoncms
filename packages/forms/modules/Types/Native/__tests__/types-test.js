import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import Form from '../../../Provider/Form';
import Text from '../Text';
import Checkbox from '../Checkbox';
// import DropZone from '../DropZone';
import Password from '../Password';
import Radio from '../Radio';
import Select from '../Select';
import Textarea from '../Textarea';

const classes = {
  formGroup: 'form-group row',
  formLabel: 'form-label col-sm-5 col',
  formControl: 'form-control col-sm-7 col',
  formError: 'small offset-sm-5',
  formText: 'small offset-sm-5 text-muted'
};

describe('Input type', () => {
  it('should render input text', () => {
    const tree = renderer
      .create(
        <Form
          onSubmit={() => {
          }}
        >
          <Text
            name={"field"}
            placeholder={"a nice placeholder"}
            {...classes}
          />
        </Form>);
    const output = tree.toJSON();
    expect(output).toMatchSnapshot();
  });

  it('should render input password', () => {
    const tree = renderer
      .create(
        <Form
          onSubmit={() => {
          }}
        >
          <Password
            name={"field"}
            placeholder={"a nice placeholder"}
            {...classes}
          />
        </Form>);
    const output = tree.toJSON();
    expect(output).toMatchSnapshot();
  });

  it('should render input textarea', () => {
    const tree = renderer
      .create(
        <Form
          onSubmit={() => {
          }}
        >
          <Textarea
            name={"field"}
            placeholder={"a nice placeholder"}
            {...classes}
          />
        </Form>);
    const output = tree.toJSON();
    expect(output).toMatchSnapshot();
  });

  it('should render input checkbox', () => {
    const tree = renderer
      .create(
        <Form
          onSubmit={() => {
          }}
        >
          <Checkbox
            name={"field"}
            placeholder={"a nice placeholder"}
            {...classes}
          >
            <option value={1}>value 1</option>
            <option value={2}>value 2</option>
            <option value={3}>value 3</option>
          </Checkbox>
        </Form>);
    const output = tree.toJSON();
    expect(output).toMatchSnapshot();
  });

  it('should render input select', () => {
    const tree = renderer
      .create(
        <Form
          onSubmit={() => {
          }}
          initialValues={{
            field: 3
          }}
        >
          <Select
            name={"field"}
            placeholder={"a nice placeholder"}
            {...classes}
          >
            <option value={1}>value 1</option>
            <option value={2}>value 2</option>
            <option value={3}>value 3</option>
          </Select>
        </Form>);
    const output = tree.toJSON();
    expect(output).toMatchSnapshot();
  });

  it('should render input select with selected option', () => {
    const tree = renderer
      .create(
        <Form
          onSubmit={() => {
          }}
          initialValues={{
            field: 3
          }}
        >
          <Select
            name={"field"}
            placeholder={"a nice placeholder"}
            {...classes}
          >
            <option value={1}>value 1</option>
            <option value={2}>value 2</option>
            <option value={3}>value 3</option>
          </Select>
        </Form>);
    const output = tree.toJSON();
    expect(output).toMatchSnapshot();
  });

  it('should render input radio', () => {
    const tree = renderer
      .create(
        <Form
          onSubmit={() => {
          }}
        >
          <Radio
            name={"field"}
            placeholder={"a nice placeholder"}
            {...classes}
          >
            <option value={1}>value 1</option>
            <option value={2}>value 2</option>
            <option value={3}>value 3</option>
          </Radio>
        </Form>);
    const output = tree.toJSON();
    expect(output).toMatchSnapshot();
  });

  it('should render input radio with checked value', () => {
    const tree = renderer
      .create(
        <Form
          onSubmit={() => {
          }}
          initialValues={{
            field: 3
          }}
        >
          <Radio
            name={"field"}
            placeholder={"a nice placeholder"}
            {...classes}
          >
            <option value={1}>value 1</option>
            <option value={2}>value 2</option>
            <option value={3}>value 3</option>
          </Radio>
        </Form>);
    const output = tree.toJSON();
    expect(output).toMatchSnapshot();
  });

  it('should render input radio with checked value `string`', () => {
    const tree = renderer
      .create(
        <Form
          onSubmit={() => {
          }}
          initialValues={{
            field: '2'
          }}
        >
          <Radio
            name={"field"}
            placeholder={"a nice placeholder"}
            {...classes}
          >
            <option value={1}>value 1</option>
            <option value={2}>value 2</option>
            <option value={3}>value 3</option>
          </Radio>
        </Form>);
    const output = tree.toJSON();
    expect(output).toMatchSnapshot();
  });

  it('should render input radio with simulated click', () => {
    const validate = jest.fn();
    const onSubmit = jest.fn();
    const component = mount(
      <Form
        onSubmit={onSubmit}
        validate={validate}
      >
        <Radio
          name={"field"}
          placeholder={"a nice placeholder"}
          {...classes}
        >
          <option value={1}>value 1</option>
          <option value={2}>value 2</option>
          <option value={3}>value 3</option>
        </Radio>
      </Form>);
    expect(component.find('input')).toHaveLength(3);
    expect(component.find('input[checked=true]')).toHaveLength(0);
    const radioInputs = component.find('input');
    radioInputs.map(input => {
      input.simulate('change');
      expect(component.find('input[checked=true]')).toHaveLength(1);
    });
    expect(validate).toHaveBeenCalledWith({ "field": 3 });
    component.simulate('submit');
    expect(onSubmit).toHaveBeenCalledWith({ "field": 3 }, expect.any(Object), expect.any(Function));
  });

  it('should render input checkbox with simulated click single value', () => {
    const validate = jest.fn();
    const onSubmit = jest.fn();
    const component = mount(
      <Form
        onSubmit={onSubmit}
        validate={validate}
      >
        <Checkbox
          name={"checkbox"}
          placeholder={"a nice placeholder"}
          {...classes}
        >
          <option value={1}>value 1</option>
          <option value={2}>value 2</option>
          <option value={3}>value 3</option>
        </Checkbox>
      </Form>);
    expect(component.find('input')).toHaveLength(3);
    expect(component.find('input[checked=true]')).toHaveLength(0);
    const checkboxes = component.find('input');
    checkboxes.map(input => {
      input.simulate('change', { target: { checked: true } });
      expect(component.find('input[checked=true]')).toHaveLength(1);
    });

    expect(validate).toHaveBeenCalledWith({ "checkbox": 3 });
    component.simulate('submit');
    expect(onSubmit).toHaveBeenCalledWith({ "checkbox": 3 }, expect.any(Object), expect.any(Function));

  });

  it('should render input checkbox with simulated click multiple values', () => {
    const validate = jest.fn();
    const onSubmit = jest.fn();
    const component = mount(
      <Form
        onSubmit={onSubmit}
        validate={validate}
      >
        <Checkbox
          multiple
          name={"checkbox"}
          placeholder={"a nice placeholder"}
          {...classes}
        >
          <option value={1}>value 1</option>
          <option value={2}>value 2</option>
          <option value={3}>value 3</option>
        </Checkbox>
      </Form>);
    expect(component.find('input')).toHaveLength(3);
    expect(component.find('input[checked=true]')).toHaveLength(0);
    const checkboxes = component.find('input');
    let cnt = 0;
    checkboxes.map(checkbox => {
      checkbox.simulate('change', { target: { checked: true } });
      expect(component.find('input[checked=true]')).toHaveLength(++cnt);
    });

    expect(validate).toHaveBeenCalledWith({ "checkbox": [1, 2, 3] });
    component.simulate('submit');
    expect(onSubmit).toHaveBeenCalledWith({ "checkbox": [1, 2, 3] }, expect.any(Object), expect.any(Function));
    checkboxes.first().simulate('change', { target: { checked: false } });
    expect(validate).toHaveBeenCalledWith({ "checkbox": [2, 3] });
    component.simulate('submit');
    expect(onSubmit).toHaveBeenCalledWith({ "checkbox": [2, 3] }, expect.any(Object), expect.any(Function));

  });
});
