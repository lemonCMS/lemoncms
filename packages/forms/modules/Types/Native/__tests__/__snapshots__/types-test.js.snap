// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text type should render input checkbox 1`] = `
<form
  className=""
  onSubmit={[Function]}
>
  <div
    className="ffc-field form-group row"
  >
    <div
      className="ffc-input form-control col-sm-7 col"
    >
      <div
        className="form-check"
      >
        <input
          checked={false}
          className="form-check-label"
          disabled={null}
          id="field_0"
          name="field"
          onChange={[Function]}
          type="checkbox"
          value={1}
        />
         
        <label
          className="form-check-label"
          htmlFor="field_0"
        >
          value 1
        </label>
      </div>
      <div
        className="form-check"
      >
        <input
          checked={false}
          className="form-check-label"
          disabled={null}
          id="field_1"
          name="field"
          onChange={[Function]}
          type="checkbox"
          value={2}
        />
         
        <label
          className="form-check-label"
          htmlFor="field_1"
        >
          value 2
        </label>
      </div>
      <div
        className="form-check"
      >
        <input
          checked={false}
          className="form-check-label"
          disabled={null}
          id="field_2"
          name="field"
          onChange={[Function]}
          type="checkbox"
          value={3}
        />
         
        <label
          className="form-check-label"
          htmlFor="field_2"
        >
          value 3
        </label>
      </div>
    </div>
  </div>
</form>
`;

exports[`Text type should render input password 1`] = `
<form
  className=""
  onSubmit={[Function]}
>
  <div
    className="ffc-field form-group row"
  >
    <input
      className="ffc-input form-control col-sm-7 col"
      name="field"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="a nice placeholder"
      type="password"
      value=""
    />
  </div>
</form>
`;

exports[`Text type should render input radio 1`] = `
<form
  className=""
  onSubmit={[Function]}
>
  <div
    className="ffc-field form-group row"
  >
    <div
      className="ffc-input form-control col-sm-7 col"
    >
      <div
        className="form-check"
      >
        <input
          checked={false}
          className="form-check-label"
          id="field_0"
          name="field"
          onChange={[Function]}
          type="radio"
          value={1}
        />
         
        <label
          className="form-check-label"
          htmlFor="field_0"
        >
          value 1
        </label>
      </div>
      <div
        className="form-check"
      >
        <input
          checked={false}
          className="form-check-label"
          id="field_1"
          name="field"
          onChange={[Function]}
          type="radio"
          value={2}
        />
         
        <label
          className="form-check-label"
          htmlFor="field_1"
        >
          value 2
        </label>
      </div>
      <div
        className="form-check"
      >
        <input
          checked={false}
          className="form-check-label"
          id="field_2"
          name="field"
          onChange={[Function]}
          type="radio"
          value={3}
        />
         
        <label
          className="form-check-label"
          htmlFor="field_2"
        >
          value 3
        </label>
      </div>
    </div>
  </div>
</form>
`;

exports[`Text type should render input radio with checked value \`string\` 1`] = `
<form
  className=""
  onSubmit={[Function]}
>
  <div
    className="ffc-field form-group row"
  >
    <div
      className="ffc-input form-control col-sm-7 col"
    >
      <div
        className="form-check"
      >
        <input
          checked={false}
          className="form-check-label"
          id="field_0"
          name="field"
          onChange={[Function]}
          type="radio"
          value={1}
        />
         
        <label
          className="form-check-label"
          htmlFor="field_0"
        >
          value 1
        </label>
      </div>
      <div
        className="form-check"
      >
        <input
          checked={true}
          className="form-check-label"
          id="field_1"
          name="field"
          onChange={[Function]}
          type="radio"
          value={2}
        />
         
        <label
          className="form-check-label"
          htmlFor="field_1"
        >
          value 2
        </label>
      </div>
      <div
        className="form-check"
      >
        <input
          checked={false}
          className="form-check-label"
          id="field_2"
          name="field"
          onChange={[Function]}
          type="radio"
          value={3}
        />
         
        <label
          className="form-check-label"
          htmlFor="field_2"
        >
          value 3
        </label>
      </div>
    </div>
  </div>
</form>
`;

exports[`Text type should render input radio with checked value 1`] = `
<form
  className=""
  onSubmit={[Function]}
>
  <div
    className="ffc-field form-group row"
  >
    <div
      className="ffc-input form-control col-sm-7 col"
    >
      <div
        className="form-check"
      >
        <input
          checked={false}
          className="form-check-label"
          id="field_0"
          name="field"
          onChange={[Function]}
          type="radio"
          value={1}
        />
         
        <label
          className="form-check-label"
          htmlFor="field_0"
        >
          value 1
        </label>
      </div>
      <div
        className="form-check"
      >
        <input
          checked={false}
          className="form-check-label"
          id="field_1"
          name="field"
          onChange={[Function]}
          type="radio"
          value={2}
        />
         
        <label
          className="form-check-label"
          htmlFor="field_1"
        >
          value 2
        </label>
      </div>
      <div
        className="form-check"
      >
        <input
          checked={true}
          className="form-check-label"
          id="field_2"
          name="field"
          onChange={[Function]}
          type="radio"
          value={3}
        />
         
        <label
          className="form-check-label"
          htmlFor="field_2"
        >
          value 3
        </label>
      </div>
    </div>
  </div>
</form>
`;

exports[`Text type should render input select 1`] = `
<form
  className=""
  onSubmit={[Function]}
>
  <div
    className="ffc-field form-group row"
  >
    <select
      className="ffc-input form-control col-sm-7 col"
      name="field"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="a nice placeholder"
    >
      <option
        selected={false}
        value={1}
      >
        value 1
      </option>
      <option
        selected={false}
        value={2}
      >
        value 2
      </option>
      <option
        selected={true}
        value={3}
      >
        value 3
      </option>
    </select>
  </div>
</form>
`;

exports[`Text type should render input select with selected option 1`] = `
<form
  className=""
  onSubmit={[Function]}
>
  <div
    className="ffc-field form-group row"
  >
    <select
      className="ffc-input form-control col-sm-7 col"
      name="field"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="a nice placeholder"
    >
      <option
        selected={false}
        value={1}
      >
        value 1
      </option>
      <option
        selected={false}
        value={2}
      >
        value 2
      </option>
      <option
        selected={true}
        value={3}
      >
        value 3
      </option>
    </select>
  </div>
</form>
`;

exports[`Text type should render input text 1`] = `
<form
  className=""
  onSubmit={[Function]}
>
  <div
    className="ffc-field form-group row"
  >
    <input
      className="ffc-input form-control col-sm-7 col"
      name="field"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="a nice placeholder"
      type="text"
      value=""
    />
  </div>
</form>
`;

exports[`Text type should render input textarea 1`] = `
<form
  className=""
  onSubmit={[Function]}
>
  <div
    className="ffc-field form-group row"
  >
    <textarea
      className="ffc-input form-control col-sm-7 col"
      name="field"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="a nice placeholder"
      rows={10}
    />
  </div>
</form>
`;
